traits/trait_list_object.py

```diff
@@ -416,6 +416,9 @@ def rename(self, name):
             self.name = name
             self.trait = trait.handler
 
+    def clear(self):
+        del self[:]
```

Comment:
I think we should still go through `list.clear` here, painful though that may be; just on general principles, each of the `TraitListObject` standard list methods should call the underlying list method. I can't *think* offhand of a case where that would matter for `clear`, but I'd prefer to be safe.